cmd/k3s/main.go

```diff
@@ -200,3 +212,33 @@ func extract(dataDir string) (string, error) {
 	}
 	return dir, os.Rename(tempDest, dir)
 }
+
+// findCriConfig returns the path to crictl.yaml
+// crictl won't search multiple locations for a config file. It will fall back to looking in
+// the same directory as the crictl binary, but that's it. We need to check the various possible
+// data-dir locations ourselves and then point it at the right one. We check:
+// - the configured data-dir
+// - the default user data-dir (assuming we can find the user's home directory)
+// - the default system data-dir
+// - the default path from upstream crictl
+func findCriConfig(dataDir string) string {
+	searchList := []string{filepath.Join(dataDir, "agent", criDefaultConfigPath)}
+
+	if homeDataDir, err := resolvehome.Resolve(datadir.DefaultHomeDataDir); err == nil {
```

Comment:
Do we want to acknowledge the error here?

Reply:
I suppose a warning would be reasonable. The only time this should ever fail is if for some reason `os.UserHomeDir()` fails in which case there are probably other things that will blow up shortly but it doesn't hurt to expose this.

Reply:
Cool cool.